app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -36,9 +41,39 @@ class ExplorationActivityTest {
     }
   }
 
+  @Test
+  fun textExplorationActivity_onBackPressed_showsStopExplorationDialog() {
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(TEST_EXPLORATION_ID_30)).use {
+      pressBack()
+      onView(withText(R.string.stop_exploration_dialog_title))
+    }
+  }
+
+  @Test
+  fun textExplorationActivity_onBackPressed_showsStopExplorationDialog_clickCancel_dismissesDialog() {
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(TEST_EXPLORATION_ID_30)).use {
+      pressBack()
+      onView(withText(R.string.stop_exploration_dialog_title))
+      onView(withText(R.string.stop_exploration_dialog_cancel_button)).inRoot(isDialog()).check(matches(isDisplayed()))
+        .perform(click())
+      it.onActivity { activity ->  not(activity.isFinishing)}
```

Comment:
This isn’t verifying anything, is it? Shouldn’t this be an assertThat to check the activity is finishing.

Reply:
@BenHenning I have tried following things, but nothing seems correct

`assertThat(explorationActivityTestRule.activity.isDestroyed).isTrue()`

`assertThat(explorationActivityTestRule.activity.isFinishing).isTrue()`

Reply:
Does the following work:

```kotlin
it.onActivity { activity -> assertThat(activity.isFinishing).isTrue() }
```

Reply:
So basically, I am not able write a proper case to test if the activity is finishing/destroyed or not.

Reply:
@BenHenning this is also not working.

Reply:
@BenHenning I am investigating this, but think it might be because after the "Leave" button is clicked we have to wait for exploration to get finished and then we are finishing the activity.

Nothing is conclusive yet.

Reply:
On further investigation, it has been confirmed that the test cases passes if we use `finish` directly and fails if we `stopExploration` and then `finish`

Reply:
This issue will be fixed in #89